server/etcdserver/api/etcdhttp/health.go

```diff
@@ -99,30 +103,53 @@ func NewHealthHandler(lg *zap.Logger, hfunc func(ctx context.Context, excludedAl
 }
 
 var (
+	// Deprecated: use healthCheckCounter instead.
 	healthSuccess = prometheus.NewCounter(prometheus.CounterOpts{
 		Namespace: "etcd",
 		Subsystem: "server",
 		Name:      "health_success",
-		Help:      "The total number of successful health checks",
+		Help:      "The total number of successful health checks (Deprecated, use etcd_server_healthchecks_total instead)",
 	})
+	// Deprecated: use healthCheckCounter instead.
 	healthFailed = prometheus.NewCounter(prometheus.CounterOpts{
 		Namespace: "etcd",
 		Subsystem: "server",
 		Name:      "health_failures",
-		Help:      "The total number of failed health checks",
+		Help:      "The total number of failed health checks (Deprecated, use etcd_server_healthchecks_total instead)",
 	})
+	healthCheckGauge = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: "etcd",
+		Subsystem: "server",
+		Name:      "healthcheck",
+		Help:      "This metric records the result of a single healthcheck.",
+	},
+		[]string{"type", "name"},
```

Comment:
```suggestion
		[]string{"path", "name"},
```

Reply:
I used `type` in order to be consistent with [K8s health metrics](https://github.com/kubernetes/kubernetes/blob/02b0b79eda5a3d41a07a6f2965ee97d492e8b6b2/staging/src/k8s.io/component-base/metrics/prometheus/slis/metrics.go#L35)